### ISSUE TITLE
put-object broken in 0.7

### ISSUE THREAD

**Issue by robertpatrick** [kordamp/oci-maven-plugin#9]
My builds which had been working perfectly are failing after picking up the latest version of the plugin:

```
[ERROR] Failed to execute goal org.kordamp.maven:oci-maven-plugin:0.7.0:put-object (upload-online-file) on project weblogic-deploy-alias-test-generate: Execution upload-online-file of goal org.kordamp.maven:oci-maven-plugin:0.7.0:put-object failed: No http provider available; add dependency on one of the oci-java-sdk-common-httpclient-* choices, e.g. oci-java-sdk-common-httpclient-jersey -> [Help 1]
```

Not sure what happened but breaking backward compatibility is not good...

**Comment by aalmiray** [comment]
The plugin was upgraded to version `3.0.1` of the oci-java-sdk. Apparently it no longer ships with a predefined http client based on that message. There's a workaround: add the given dependency as suggested to the build's classpath, such as

**settings.gradle**
```groovy
buildscript {
    repositories {
        mavenCentral()
        gradlePluginPortal()
    }
    dependencies {
        classpath 'com.oracle.oci.sdk:oci-java-sdk-common-httpclient-jersey:3.0.1'
    }
}
```

If additional dependencies may be required then they may be added in the same way. This is the standard Gradle way to do so.

Also, we reserve the right to break compatibility during the `0.x` series (although it's not our intention to break it willy-nilly). BC will be guaranteed once the plugin graduates to 1.0.0, we'll follow semver.

**Comment by robertpatrick** [comment]
Seems like the plugin should ship with a provider.  Not sure why I, as a user of the plugin, needs to understand its dependencies and account for missing ones.  I guess I can just build my own plugin to eliminate this sort of churn...

**Comment by aalmiray** [comment]
Yes, the plugin could ship with it's own provider in the future, as it did in the past when a provider was bundled in. That is, the plugin comes with batteries included. We'll get it ready for the next release.

**Comment by aalmiray** [comment]
Fixed in `v0.8.0`.
